python/bigipconfigdriver.py

```diff
@@ -401,6 +372,53 @@ def _do_reset(self):
         if self._interval:
             self._interval.stop()
 
+    def _update_cccl(self, config):
+        # customProfiles is true when we've written out a custom profile.
+        # Once we know we've written out a profile, we can call delete
+        # if needed.
+        customProfiles = False
+
+        _handle_vxlan_config(config)
+        cfg_net = create_network_config(config)
+        incomplete = 0
+        for mgr in self._managers:
+            partition = mgr.get_partition()
+            cfg_ltm = create_ltm_config(partition, config)
+            try:
+                # Manually create custom profiles;
+                # CCCL doesn't yet do this
+                if 'customProfiles' in cfg_ltm and \
+                        mgr.get_schema_type() == 'ltm':
+                    tmp = 0
+                    customProfiles, tmp = _create_custom_profiles(
+                        mgr.mgmt_root(),
+                        partition,
+                        cfg_ltm['customProfiles'])
+                    incomplete += tmp
+
+                # Apply the BIG-IP config after creating profiles
+                # and before deleting profiles
+                if mgr.get_schema_type() == 'net':
+                    incomplete += mgr._apply_net_config(cfg_net)
+                else:
+                    incomplete += mgr._apply_ltm_config(cfg_ltm)
+
+                # Manually delete custom profiles (if needed)
+                if customProfiles and \
+                        mgr.get_schema_type() == 'ltm':
+                    _delete_unused_ssl_profiles(
+                        mgr.mgmt_root(),
+                        partition,
+                        cfg_ltm)
+
+            except F5CcclError as e:
```

Comment:
The stacktrace I got from Mark's runs showed the last frame raising ConnectionError. Are we sure that will be converted to an F5CcclError? We may need to catch ConnectionError as well.

His stacktrace is in the comment of issue 506.